### PR TITLE
Add search functionality to /blog

### LINKED ISSUE #25: Implement search on the blog page
## This issue has a bounty of $5. Please read the [contribution guide](https://github.com/slightlynerd/startupgraveyard.africa/blob/main/CONTRIBUTION.md) for instructions and terms of bounties.

**Is your feature request related to a problem? Please describe.**
There's no way to search through blog articles.

**Describe the solution you'd like**
Add a search input field to the blog page (https://startupgraveyard.africa/blog) and implement the search functionality.
- search input should be above the "Recent Posts" text
- it should be centered and 50% width on desktop, 80% on tablet, and 100% on mobile.
- search lookup is not triggered until the user enters at least 2 characters
- search results should show the top 4 results in descending order, from the most recent to the oldest

### PR DESCRIPTION
**Pull request by tsomic**
Closes #25 

I did not take the pagination into account, but at the moment that is not properly implemented anyways.

### CODE REVIEW

File: pages/blog/index.vue [hunk 1]
```diff
@@ -56,10 +66,21 @@ const totalCount = ref<number>(1);
...
 // computed
 const pageCount = computed(() => Math.ceil(totalCount.value / DEFAULT_PAGE_SIZE));
 
+const filteredBlogs = computed(() => {
+  if (!searchTerm.value || searchTerm.value.length < 2) {
+    return recentBlogPosts.value;
+  }
+
+  return recentBlogPosts.value.filter(blog =>
+    blog.title.toLowerCase().includes(searchTerm.value.toLowerCase())
```
Comment (KelvinGoboZap) [inline pages/blog/index.vue:80]: @tsomic I see you only implemented a client-side search as you observed that the pagination is broken. I've resolved the bug with the pagination so pull the changes and make the search work with data from the server.

**Review by KelvinGoboZap — CHANGES_REQUESTED** [review]

**Comment by KelvinGoboZap** [comment]
@tsomic I noticed the pagination bug was introduced in your previous PR (https://github.com/slightlynerd/startupgraveyard.africa/pull/27) which was causing the `onPaginationChanged` handler to be called twice. I've resolved the bug so pagination should be working and you should be able to complete the search functionality.

Also, I notice a typo in the PR title. **earch** should be **search**.

File: package.json [hunk 2]
```diff
@@ -13,31 +13,31 @@
     "start": "node .output/server/index.mjs"
   },
   "dependencies": {
-    "@nuxtjs/sitemap": "^5.1.0",
+    "@nuxtjs/sitemap": "^5.1.4",
```
Comment (slightlynerd) [inline package.json:16]: @tsomic What's the reason for updating the packages? Just want to understand why this was done.

File: pages/blog/index.vue (hunk 1 above)
Comment (slightlynerd) [inline pages/blog/index.vue:80]: @tsomic I see you only implemented a client-side search as you observed that the pagination is broken. I've resolved the bug with the pagination so pull the changes and make the search work with data from the server.

**Review by slightlynerd — CHANGES_REQUESTED** [review]
@tsomic I noticed the pagination bug was introduced in your previous PR (https://github.com/slightlynerd/startupgraveyard.africa/pull/27) which was causing the `onPaginationChanged` handler to be called twice. I've resolved the bug so pagination should be working and you should be able to complete the search functionality.

Also, I notice a typo in the PR title. earch should be search.

File: package.json (hunk 2 above)
Comment (tsomic) [reply package.json:16]: Just to keep packages updated regularly, no urgent reason. Of course I can revert that commit if you prefer

Comment (slightlynerd) [reply package.json:16]: @tsomic It's fine. Double-check to confirm existing functionality is not broken.

File: pages/blog/index.vue (hunk 1 above)
Comment (tsomic) [reply pages/blog/index.vue:81]: I have some problems trying to implement the search with data from the server. This is what I have been trying and I don't receive any data. 
image
https://stackoverflow.com/questions/46568142/google-firestore-query-on-substring-of-a-property-value-text-search

And even if it worked, I am pretty sure the search would be case-sensitive with the current configuration in Firebase. Do you have any ideas?

Comment (slightlynerd) [reply pages/blog/index.vue:81]: @tsomic I see the issue. We would have to use a temporary workaround and revert to client-side pagination and search. 

So basically what would be done is that all articles are fetched on the blog page. So pagination and search would be done client-side.

A `TODO:` comment should be added to the method for client-side pagination so it's refactored to server-side search and pagination at a later time.

Let me know if you can work on this or I can take it.

Comment (tsomic) [reply pages/blog/index.vue]: I added the comment and the client-side pagination and search.

I would prefer if you could work on the server-side implementation :)

**Review by slightlynerd — APPROVED** [review]

File: pages/blog/index.vue (hunk 1 above)
Comment (slightlynerd) [reply pages/blog/index.vue]: @tsomic Yes, I'll work on the server-side implementation at a later time :)

**Pull request merged**